app/ydoc-server/src/languageServerSession.ts

```diff
@@ -570,7 +587,23 @@ class ModulePersistence extends ObservableV2<{ removed: () => void }> {
           syncModule.applyEdit(edit)
         } else {
           const { root, spans } = Ast.parseModuleWithSpans(code, syncModule)
```

Comment:
When we use the snapshot (and `syncToCode(code)`), there's no need to parse `code`. The process should look like:
- `root = parse(snapshotCode ?? code)`; `syncModule.syncRoot(root)`; attach IDs.
- If there is snapshot and code's value is different from it, `root.syncToCode(code)`